tokio-trace-tower-http/src/lib.rs

```diff
@@ -9,38 +9,56 @@ extern crate tokio_trace_futures;
 use std::marker::PhantomData;
 
 use futures::{Future, Poll};
-use tokio_trace::{field, Level, Span};
+use tokio_trace::{field, Span};
 use tokio_trace_futures::{Instrument, Instrumented};
 use tower::MakeService;
 use tower_service::Service;
 
 #[derive(Debug)]
 pub struct InstrumentedHttpService<T> {
     inner: T,
-    span: Span,
+    span: Option<Span>,
 }
 
 impl<T> InstrumentedHttpService<T> {
-    pub fn new(inner: T, span: Span) -> Self {
-        Self { inner, span }
+    pub fn new(inner: T) -> Self {
+        Self { inner, span: None }
+    }
+
+    pub fn with_span(inner: T, span: Span) -> Self {
+        Self {
+            inner,
+            span: Some(span),
+        }
     }
 }
 
 #[derive(Debug)]
 pub struct InstrumentedMakeService<T, B> {
     inner: T,
-    span: Span,
+    span: Option<Span>,
     _p: PhantomData<fn() -> B>,
 }
 
 impl<T, B> InstrumentedMakeService<T, B> {
-    pub fn new<Target>(inner: T, span: Span) -> Self
+    pub fn new<Target>(inner: T) -> Self
+    where
+        T: MakeService<Target, http::Request<B>>,
+    {
+        Self {
+            inner,
+            span: None,
+            _p: PhantomData,
+        }
+    }
+
+    pub fn with_span<Target>(inner: T, span: Span) -> Self
```

Comment:
Consider taking an `Into<Option<Span>>` instead of a `Span` here
```suggestion
    pub fn with_span<Target>(inner: T, span: impl Into<Option<Span>>) -> Self
```

That way, this function can be called with either a `span` _or_ a `Some(span)`, which might make the code in `poll` for `InstrumentedMakeServiceFuture` a little simpler (and make the API more flexible)